### PR TITLE
Removed Hardcoded Camera3D Orthographic Size Limit

### PR DESCRIPTION
**Pull request by iiMidknightii**
Implements [5736](https://github.com/godotengine/godot-proposals/issues/5736).  Removes the hard-coded upper limit for Camera3D:set_size(). Updated property editor hint to allow higher values as well.

### CODE REVIEW

**Comment by clayjohn** [comment]
It looks like the clamps were added in https://github.com/godotengine/godot/pull/34665 to avoid the gizmo setting values way outside the ideal range. This PR will reintroduce the problem that https://github.com/godotengine/godot/pull/34665 was designed to solve. @timothyqiu any thoughts? Is there a way for us to clamp gizmo input while still allowing using to set a wider range if they need?

**Comment by timothyqiu** [comment]
#34665 mainly solves the lower bound issue, so expanding the upper bound won't affect it.

https://github.com/godotengine/godot/blob/d171a2d7bd44fb0e4b4fd7ddbce7d1611278724e/editor/plugins/node_3d_editor_gizmos.cpp#L1783-L1792

Since the `get_closest_points_between_segments()` call uses a max distance of 4096, `d` can reach 8192 at most.

It was a mistake to use `CLAMP()` instead of a simple `MAX()` here. To sync the gizmo range with the slider range, I think you can change 4096 on line 1784 to `16384 / 2`.

File: scene/3d/camera_3d.cpp [hunk 1]
```diff
@@ -591,7 +591,7 @@ void Camera3D::set_fov(real_t p_fov) {
 }
 
 void Camera3D::set_size(real_t p_size) {
-	ERR_FAIL_COND(p_size < 0.001 || p_size > 16384);
+	ERR_FAIL_COND(p_size <= 0.0);
```
Comment (aaronfranke) [inline scene/3d/camera_3d.cpp:594]: As a sanity check, we may want to make the lower limit be `CMP_EPSILON` instead of `0.0`.

**Review by aaronfranke — COMMENTED** [review]
Makes sense to me, aside from the comment below, and the commits need to be squashed into one.

**Pull request closed without merging**